File: bundle.py
source_folder = "src/"
build_folder = "build/"

wrapper_path = source_folder + "wrapper.html"
content_tag = "<!-- Content -->"

content_path = build_folder + "content.html"

output_path = build_folder + "index.html"

# Get Wrapper
wrapper_file = open(wrapper_path, "r")
wrapper = wrapper_file.read()
wrapper_file.close()

# Get content
content_file = open(content_path, "r")
content = content_file.read()
content_file.close()

# Inject content
output_html = wrapper.replace(content_tag, content)

# Write output
output_file = open(output_path, "w")
output_file.write(output_html)
output_file.close()

print("Bundled " + content_path + " into " + output_path)

File: bundle.js
const fs = require("fs");

const wrapperPath = "build/wrapper.html";
const inputContentPath = "build/content.html";
const contentTag = "<!-- Content -->";

const outputPath = "build/bundled.html";

// Get Wrapper
let wrapperHtml = fs.readFileSync(wrapperPath).toString();

let contentHtml = fs.readFileSync(inputContentPath).toString();

wrapperHtml = wrapperHtml.replace(contentTag, contentHtml);

fs.writeFile(outputPath, wrapperHtml, function (err) {
    if (err) return console.log(err);
    console.log(`Bundled site into ${outputPath}`);
});
